### PR TITLE
Add CSS style to messages from same user (from @xPaw)

### PR DESCRIPTION
**Pull request by DanielOaks**
Essentially, this PR adds the `equal-from` class to subsequent messages from the same user. For example:

lounge-equal-from

This lets themes use special formatting of these messages. For instance:

lounge-example-display

Almost all the code for this came from @xPaw in #659, I just did some minor updates so it'd load and work correctly on my box. I haven't yet played with loading history, but I'll take a squiz at that and see what I can find.

**edit:** This is my first dive into node stuff, so feel free to rip apart any changes I've made :P

### CODE REVIEW

**Comment by DanielOaks** [comment]
19231cd7b486fee58ff0501cb18fb8dd22214d73 adds the `equal-from` class appropriately when loading history

**Comment by xPaw** [comment]
Maybe as a default we would fade out these nicks to show to users that we have such a feature?

**Comment by DanielOaks** [comment]
I imagine you could probably fade them out without an issue in the CSS yeah, just CSS animate the height/width/opacity and all, let the theme change it from one to the other?

**Comment by xPaw** [comment]
```.message.equal-from {
opacity: .7;
}```

would probably do the trick, with optional `opacity:1` on hover.

File: client/js/lounge.js [hunk 1]
```diff
@@ -326,8 +335,16 @@ $(function() {
...
+		var channel = chat.find("#chan-" + data.id + " .messages");
 		var documentFragment = buildChannelMessages(data.id, data.messages);
-		var channel = chat.find("#chan-" + data.id + " .messages").append(documentFragment);
+
+		channel.append(documentFragment);
+
+		if (data.messages.length > 0) {
+			var lastSender = data.messages[data.messages.length - 1];
+			lastSender = lastSender.mode + lastSender.from;
+			channel.data("last-from", lastSender);
```
Comment (astorije) [inline client/js/lounge.js]: Why this back-and-forth between storing in JS (`lastSender`) and storing in DOM (`data-last-from`)?

Isn't there a better way to do that, like looking at who the last message is from directly without storing that `data`?

File: client/js/lounge.js [hunk 2]
```diff
@@ -452,6 +471,13 @@ $(function() {
 			.find("#chan-" + data.chan)
 			.find(".messages");
 
+		// set lastSender appropriately
```
Comment (astorije) [inline client/js/lounge.js]: Probably needs a more descriptive comment.

File: client/views/msg.tpl [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div class="msg {{type}}{{#if self}} self{{/if}}{{#if highlight}} highlight{{/if}}" id="msg-{{id}}" data-time="{{time}}">
+<div class="msg {{type}}{{#if self}} self{{/if}}{{#if highlight}} highlight{{/if}}{{#if sameSender}} equal-from{{/if}}" id="msg-{{id}}" data-time="{{time}}">
```
Comment (astorije) [inline client/views/msg.tpl]: I'd prefer `same-sender` or `same-from` than `equal-from`. Thoughts?

File: client/views/msg_action.tpl [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<div class="msg {{type}}{{#if self}} self{{/if}}{{#if highlight}} highlight{{/if}}" id="msg-{{id}}" data-time="{{time}}">
+<div class="msg {{type}}{{#if self}} self{{/if}}{{#if highlight}} highlight{{/if}}{{#if sameSender}} equal-from{{/if}}" id="msg-{{id}}" data-time="{{time}}">
```
Comment (astorije) [inline client/views/msg_action.tpl]: I don't think we want that on actions, what's the use case? "Condensing" senders between normal messages and action messages would be unfortunate.

**Comment by astorije** [comment]
> Maybe as a default we would fade out these nicks to show to users that we have such a feature?

This risks to be confusing for new users, like thinking messages are not being sent (offline support) yet or something. These will need to be documented at some point, and that's my next contribution once I'm done catching up with PR reviews.

**Comment by astorije** [comment]
@DanielOaks, ping? Could you rebase on `master` and fix conflicts please? Thanks 😃

**Comment by astorije** [comment]
Woops, merged instead of rebased, @xPaw 😅 (I'm thinking you probably did that on purpose to not affect the history, but we'll still have to clean this up at some point).

@DanielOaks, do you mind addressing my comments if you can? Thanks!

**Comment by christer88** [comment]
@astorije "This risks to be confusing for new users, like thinking messages are not being sent (offline support) yet or something." I do see your point, however, this is a feature in IRCCloud and it will make the client a lot better on small screens with less nicks and more text.

I hope @DanielOaks have time to continue on this PR, or let someone else take over. Thank you in advance :)

**Comment by astorije** [comment]
> I do see your point, however, this is a feature in IRCCloud and it will make the client a lot better on small screens with less nicks and more text.

Sorry if I was unclear, I didn't mean that entire feature/capability would be confusing, I meant fading out messages would be.

**Comment by astorije** [comment]
@DanielOaks, do you want to pick this up and fix conflicts? I know this PR is a decade old, so I'd understand if you'd rather have someone from @thelounge/maintainers to take it from here.

**Comment by astorije** [comment]
Hey @DanielOaks! We are going to close this PR for now, feel free to reopen (or ask to reopen it if you can't) if you are planning to pick up the work on this.
Unfortunately, it looks like none of the @thelounge/maintainers has the time to take over this for now, but any help from the community will be welcome!

**Comment by christer88** [comment]
@DanielOaks Can you please continue on this PR? It would be very nice to have this feature in TheLounge, especially on small screens where nick and time takes a lot of space. Thank you so much in advance!

**Comment by DanielOaks** [comment]
Ah, I'm very sorry for not responding @astorije! I don't really use Lounge these days so I can't promise anything, but I'll keep it on my radar for if I pickup other work again (if anyone else wants to pick it up, please feel free to go ahead!) Sorry for the trouble @christer88 😛

**Comment by astorije** [comment]
@christer88, would you like to give it a shot?

**Pull request closed without merging**